tests/framework/crossplane.go

```diff
@@ -0,0 +1,218 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+// ExpectedNginxField contains an nginx directive key and value,
+// and the expected file, server, and location block that it should exist in.
+type ExpectedNginxField struct {
+	// Key is the directive name.
+	Key string
+	// Value is the value for the directive. Can be the full value or a substring.
```

Comment:
If a substring, does `ValueSubstringAllowed` need to be true? If so, let's add a comment.